aptos-move/framework/aptos-stdlib/sources/table_with_length.spec.move

```diff
@@ -0,0 +1,79 @@
+/// Specifications of the `table_with_length` module.
+spec aptos_std::table_with_length {
+
+    // Make most of the public API intrinsic. Those functions have custom specifications in the prover.
+
+    spec TableWithLength {
+        pragma intrinsic = map,
+            map_new = new,
+            map_destroy_empty = destroy_empty,
+            map_len = length,
+            map_is_empty = empty,
+            map_has_key = contains,
+            map_add_no_override = add,
+            map_add_override_if_exists = upsert,
+            map_del_must_exist = remove,
+            map_borrow = borrow,
+            map_borrow_mut = borrow_mut,
+            map_spec_get = spec_get,
+            map_spec_set = spec_set,
+            map_spec_del = spec_remove,
+            map_spec_len = spec_len,
+            map_spec_has_key = spec_contains;
+    }
+
+    spec new {
+        pragma intrinsic;
+    }
+
+    spec destroy_empty {
+        pragma intrinsic;
+    }
+
+    spec length {
+        pragma intrinsic;
+    }
+
+    spec empty {
+        pragma intrinsic;
+    }
+
+    spec add {
+        pragma intrinsic;
+    }
+
+    spec borrow {
+        pragma intrinsic;
+    }
+
+    spec borrow_mut {
+        pragma intrinsic;
+    }
+
+    spec borrow_mut_with_default {
+        pragma opaque, verify=false;
+        aborts_if false;
+        // TODO: Prover need to extend with supporting the `borrow_mut_with_default` pragma.
+        // `table.length` cannot be accessed because the struct is intrinsic.
```

Comment:
`table.length` cannot be accessed but the length can be obtained with `spec_len`

Reply:
Yeah, you're right. I think we can specify about the change of length using `spec_len`.
However, there is another difficulty (which I should've put in the comment). It's difficult to write the postcondition about the mutable reference to return. If the key doesn't exist, it inserts the default value to the table, and return its mutable reference. I am not sure if we can specify it precisely as the spec of an opaque function.